### DISCUSSION TITLE
Using raygui in large C++ project.

### DISCUSSION THREAD

**Discussion by SourceOfDeath** [raysan5/raylib#3772] [Q&A] [answered]
I am using raylib + raygui in a c++ project consisting of several files and classes. I include libraries in the main *.cpp file:
`#include "raylib.h"`
`#define RAYGUI_NO_ICONS`
`#define RAYGUI_IMPLEMENTATION`
`#include "raygui.h"`
I want to call different functions from libraries from different project files. 
I want to use the GetTextWidth("blahblahblah") function. This is possible in the main source code file, but a compilation error occurs in the rest of the project files, where library headers are included.
`error C3861: GetTextWidth: id not found` (Microsoft Visual Studio Community 2019 16.11.26)
I have reviewed the raygui.h file.
`static int GetTextWidth(const char *text)`
So the compiler is right. A static function in C is a function that has a scope that is limited to its object file. This is the file in which 
`#define RAYGUI_IMPLEMENTATION`
`#include "raygui.h"`
that is, the main file.
However, many functions are still available in other files where the library is included. For example, GuiButton, GuiLabel, GuiWindowBox and others. But they all get the Rectangle parameter setting their size and location. To do this, I would like to be able to call GetTextWidth everywhere. Please tell me the right way to do this in Microsoft Visual Studio Community 2019 16.11.26.

**Comment by raysan5** [comment, marked as answer]
@SourceOfDeath I'm afraid that function is not exposed, you can use the raylib equivalent `MeasureTextEx()` or just expose it for your needs.

**Reply by SourceOfDeath** [reply]
Thanks. Using a function from raylib is a good solution.

**Comment by SourceOfDeath** [comment]
I created a structure with static fields to store pointers to functions. These fields are filled in in the main file at startup. And the structure itself is then available everywhere. It works, but it looks weird.
